Makefile

```diff
@@ -1,116 +0,0 @@
-.PHONY: clean-pyc clean-build docs clean docs
```

Comment:
Why are you removing this? This is pretty handy to have...

Reply:
everything should be done using the poetry CLI now so the purpose here is to eliminate a tool that's not needed anymore.

Reply:
I don't think that the need for this file has gone away. Some of this file can go away, sure. But this file gave a very simple interface for a lot of common functionality. We can still have this file invoke `poetry run ....` but I'd rather keep the makefile.

Reply:
lets replace the makefile with [poetry aliases](https://pypi.org/project/poetry-aliases). this PR is already pretty large so lets do it as a follow on PR?

Reply:
Poetry aliases are not project specific. It appears they are more like git aliases and set up globally. They won't work for this use case.